docs/proposals/2017-5-26 - Advanced Table.md

```diff
@@ -0,0 +1,446 @@
+
+* made up of 4 objects - `advanced_table`, `row`, `data_cell`, `header_cell`
+  * a `table` is a list of rows
+  * a `row` is a list of cells
+  * a `cell` is a list of content elements, restricted to text and raw_html
+    * there are two implementations of a cell, `data_cell` and `header_cell`, similar to `<th>` and `<td>` tags
+* order/layout assumptions - elements will have an assumed layout that clients may follow
+  * for example, rows are ordered from top to bottom, and cells are ordered left to right
+* no other rendering details will be included in the spec
+  * if needed, adapters can use additional properties to specify rendering details
+
+
+## Advanced Table
+
+`advanced_table` is the parent object. It is a content element, and contain a list of rows.
+
+### Rows
+
+List of rows that belong to the table. The order is assumed to be top to bottom.
+
+### Title
+
+Text field. Name of the table.
+
+### Citation
+
+ANS text element. The source of the table, for example "The Washington Post".
+
+### Credits
+
+ANS credits object, for identifying the author(s) of the table.
+
+### Subtype
+
+ANS subtype. Can be used to distinguish between different types of tables, so that clients can react accordingly.
+
+### No height/width or other display options
+
+These are intentionally left out.
+
+If adapters need to include any style or rendering details, they can use additional properties on any component of a table - table, row, cell, or content element.
+
+
+## Row
+
+A `row` contains a list of cells. It will only be usable within a table.
+
+### Cells
+
+List of cells that belong to this row. The order is assumed to be left to right.
+
+### Subtype
+
+ANS Subtype. Can be used to distinguish types of rows.
+
+## Data Cell / Header Cell
+
+There are two types of cells, `data_cell` and `header_cell`. They will only be usable within a row.
+
+Both `header_cell` and `data_cell` will contain a list of content elements. They have the same spec, except for the type.
+
+### Cell Elements
+
+A list of content elements that belong to the cell. This is restricted to elements of type `text` and `raw_html`.
+
+The order of elements is assumed to be top to bottom.
+
+### Subtype
+
+ANS Subtype. Can be used to distinguish types of cells.
+
+
+## Alternatives
+
+### Cell types
+
+The spec could potentially have one type for cells, `cell`, and use subtype to distinguish between `data` and `header` cells. This would make the schema smaller, but would also require clients check two fields on each cell. I don't have strong feelings on this either way.
+
+
+## Concerns
+
+### What happens if rows have different numbers of cells?
+
+Based on the order of elements and the assumed layout (left to right, or top to bottom), it is clear where each element should go. If the user wants to leave empty space, they would include empty elements. Otherwise, it will be assuemd that any mismatch in length will lead to space at the end of the element.
```

Comment:
To make sure I understand this:

Row 1 has 2 cells, 
Row 2 has 5 cells

When rendering, the cell 2 in row 1 should have the same width as the total width of cells 2-5 in row 2

Is that correct?